### PR TITLE
Add one-to-many join bidirectional between branch and chain

### PR DESCRIPTION
**Pull request by RehasLiamsi**

### CODE REVIEW

**Comment by RehasLiamsi** [comment]
Linked with #36 #37 #70

**Comment by RehasLiamsi** [comment]
@figgge can you please take a look at the failing tests?

File: src/main/java/com/example/springbootproject/controller/ChainController.java [hunk 1]
```diff
@@ -56,67 +66,93 @@ List<ChainName> getAllDtoNames() {
...
-
+    @PutMapping("{chainId}/stores/{storeId}")
+    @Transactional
+    public void addStoreToChain(@PathVariable Long chainId, @PathVariable Long storeId) {
+            Store store = storeRepository.findById(storeId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+            Chain chain = chainRepository.findById(chainId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+            chain.getStores().add(store);
 }
+
+    @DeleteMapping("{chainId}/stores/{storeId}")
```
Comment (Lugn1) [inline src/main/java/com/example/springbootproject/controller/ChainController.java:144]:
```suggestion
    @DeleteMapping("/{chainId}/stores/{storeId}")
```

File: src/main/java/com/example/springbootproject/controller/ChainController.java [hunk 2]
```diff
@@ -56,67 +66,93 @@ List<ChainName> getAllDtoNames() {
...
+                .ifPresent(chain -> chain.getStores().removeIf(store -> store.getId().equals(storeId)));
+    }
+
+
+    @DeleteMapping("{chainId}/stores/{storeId}")
+    @Transactional
+    public void deleteBranchFromChain(@PathVariable Long chainId, @PathVariable Long storeId) {
+        chainRepository.findById(chainId)
+                .ifPresent(chain -> chain.getStores().removeIf(store -> store.getId().equals(storeId)));
+    }
```
Comment (figgge) [inline src/main/java/com/example/springbootproject/controller/ChainController.java:157]:
```suggestion
   
```

File: src/main/java/com/example/springbootproject/controller/ChainController.java (hunk 1 above)
Comment (Lugn1) [inline src/main/java/com/example/springbootproject/controller/ChainController.java:144]:
```suggestion
    @DeleteMapping("/{chainId}/stores/{storeId}")
```

We need the forward slash before {chainId)

File: src/main/java/com/example/springbootproject/controller/ChainController.java [hunk 3]
```diff
@@ -56,67 +66,93 @@ List<ChainName> getAllDtoNames() {
...
+
+    @DeleteMapping("{chainId}/stores/{storeId}")
+    @Transactional
+    public void deleteStoreFromChain(@PathVariable Long chainId, @PathVariable Long storeId) {
+        chainRepository.findById(chainId)
+                .ifPresent(chain -> chain.getStores().removeIf(store -> store.getId().equals(storeId)));
+    }
+
+
+    @DeleteMapping("{chainId}/stores/{storeId}")
```
Comment (Lugn1) [inline src/main/java/com/example/springbootproject/controller/ChainController.java:152]:
```suggestion
    @DeleteMapping("/{chainId}/stores/{storeId}")
```

File: src/main/java/com/example/springbootproject/controller/StoreController.java [hunk 4]
```diff
@@ -57,25 +61,37 @@ void addStore(@RequestBody Store store) {
...
 
     }
+
+    @PutMapping("{storeId}/chains/{chainId}")
+    @Transactional
+    public void addChainToStore(@PathVariable Long storeId, @PathVariable Long chainId){
+        Store store = storeRepository.findById(storeId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+        chainRepository.findById(chainId).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+        chainRepository.findChainById(chainId).getStores().add(store);
+    }
```
Comment (figgge) [inline src/main/java/com/example/springbootproject/controller/StoreController.java:96]:
```suggestion
  ```
We should remove this unused method.

**Review by Lugn1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/springbootproject/controller/StoreController.java (hunk 4 above)
Comment (RehasLiamsi) [reply src/main/java/com/example/springbootproject/controller/StoreController.java:96]: Hi, I added this method to be able to add chain to an existing store through the store database as well. That is why I put this method in store. If you think this is unnecessary and it should suffice with just being able to add stores to chains through the chain database, I can delete this method.

**Review by figgge — APPROVED** [review]

**Review by Lugn1 — APPROVED** [review]

**Pull request merged**